tools/klee-zesti/klee-zesti

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env python3
+import sys
+import os
+import subprocess
+import tempfile
+import select
+HELP="""OVERVIEW: ZESTI like wrapper of KLEE
+
+USAGE:  klee-zesti [klee-options] <input bytecode> <concrete program arguments>
+
+WARNING this script is not eqvivalent to ZESTI in ICSE 2012. It just provides a similar interface to klee. Namely it first explores the path of <concrete prgoram arguments> and then continues symbolic execution from that point. Most importantly it does not implement the ZESTI searcher.
+"""
+
+
+#GEN_BOUT="/data/klee/build7.0/bin/gen-bout"
+#KLEE="/data/klee/build7.0/bin/klee"
+KLEE="klee"
+GEN_BOUT="gen-bout"
+
+def find_klee_bin_dir():
+  global KLEE
+  global GEN_BOUT
+  bin_dir = os.path.dirname(os.path.realpath(__file__))
```

Comment:
This would look nicer (and be more portable) with `pathlib`.

Reply:
It wouldn't be nicer and we don't usppoort windows anyway

Reply:
Why? Wouldn't it boil down to `Path(__file__).resolve().parent` + nicer append?

Reply:
And then call a function everytime i want `subprocess.run`?

Reply:
You mean `str()`? Just change the type here.

Reply:
So calling 6 functions (Path, resolve, parent and 3x str) is nicer than calling 2?

Reply:
Yes, it's nicer than error-prone string concatenation and low-level os.path handling. At least it has a proper type...

Reply:
You call it proper type, I would call it wrong type, considering you can't use it anywhere

Reply:
Path handling + isfile is somewhere. Just the subprocess API is limited...

Reply:
But the `subprocess` API is the core of this script, everything else should be in its service.